packages/vite/src/node/config.ts

```diff
@@ -1039,7 +1039,11 @@ async function loadConfigFromBundledFile(
     try {
       return (await dynamicImport(fileUrl)).default
     } finally {
-      fs.unlinkSync(fileNameTmp)
+      try {
+        fs.unlinkSync(fileNameTmp)
+      } catch {
+        // already removed if this function is called twice simultaneously
+      }
```

Comment:
Nice! This could explain the flakiness we experienced after the file name scheme change. I think there may not be possible to get a race condition between writing the file and doing the dynamic import, so this fix should work. Let's merge this one for now.

But maybe we should directly make sure we generate a unique name? Looks like we don't have a way to do UUIDs right now, so we could use a simple scheme as we do here https://github.com/vitejs/vite/blob/c1fa219f7da09d8fca0140c79e62dbce6d54b0e5/packages/vite/src/node/ssr/ssrTransform.ts#L88 plus the current timestamp. I'm not sure if it is needed though, reusing the same file looks ok to me.

Reply:
Yeah I think this should now work fine too as it's not possible for two simultaneous call to have one unlinking first before dynamic importing. Though I'd agree that a unique name would be a lot safer if we implement one day.